www/source/blog/2018-08-18-lift-and-shift-hab.html.md

```diff
@@ -0,0 +1,396 @@
+
+## Prerequisites for Kubernetes cluster deployment 
+
+Transitioning to a Kubernetes cluster at this point might seem like a big leap but to our advantage we have already exported the required Habitat package artifacts.
+ 
+In the next two sections we will be deploying to [Google Kubernetes Engine (GKE)](https://cloud.google.com/kubernetes-engine/) with [Habitat Operator](https://github.com/habitat-sh/habitat-operator) already installed.  Similarly to the steps described [here](https://www.habitat.sh/blog/2018/07/habitat-gcr-and-gke/) we will upload our artifacts to [Google Container Registry](https://cloud.google.com/container-registry/) for convenience in our manifest.  As an example, the following commands would upload our exported Table Setting image to GCR:
+
+```bash
+# assumes docker and gcloud are installed locally
+$ gcloud auth configure-docker
+$ docker tag habskp/table-setting:latest eu.gcr.io/spaterson-project/table-setting:latest
+$ docker push eu.gcr.io/spaterson-project/table-setting:latest
+```
+
+Also worthwhile to comment that although GCP was chosen for this post, the same artifacts could be deployed in the same way to other Cloud provider container registries and Kubernetes clusters simply by supplying the appropriate deployment manifest files. 
+
+## Deploying to a Habitat-enabled Kubernetes cluster
+
+The below manifest deploys the Habitat managed MySQL and Table Setting images uploaded to [Google Container Registry](https://cloud.google.com/container-registry/).  Similar environment variables as before are set to configure the database:
+
+```bash
+$ cat gke-gcr-hab-core-mysql-sample.yml
+---
+apiVersion: habitat.sh/v1beta1
+kind: Habitat
+metadata:
+  name: database
+  labels:
+    app: table-setting-hab-db
+customVersion: v1beta2
+spec:
+  v1beta2:
+    image: eu.gcr.io/spaterson-project/mysql:latest
+    count: 1
+    env:
+      - name: HAB_MYSQL
+        value: "bind = '0.0.0.0'\napp_username = 'tablesetting'\napp_password = 'tablesetting'\nroot_password = 'notverysecure'"
+    service:
+      name: mysql
+      topology: standalone
+---
+apiVersion: habitat.sh/v1beta1
+kind: Habitat
+metadata:
+  name: table-setting-app
+  labels:
+    app: table-setting-hab
+customVersion: v1beta2
+spec:
+  v1beta2:
+    image: eu.gcr.io/spaterson-project/table-setting:latest
+    count: 1
+    env:
+      - name: DB_TYPE
+        value: mysql
+      - name: DB_USER
+        value: tablesetting
+      - name: DB_PASSWD
+        value: tablesetting
+    service:
+      name: table-setting
+      topology: standalone
+      bind:
+        - name: database
+          service: mysql
+          group: default
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: table-setting-lb
+spec:
+  type: LoadBalancer
+  selector:
+    habitat-name: table-setting-app
+  ports:
+  - port: 80
+    targetPort: 5000
+```
```

Comment:
Why aren't you using the Kubernetes exporter?

Reply:
I wanted to illustrate that the same artifact runs locally with docker-compose and on GKE.

Reply:
For future reference, the Kubernetes exporter basically runs the same `hab pkg export docker` command to create an image, but also generates Kubernetes YAML configuration files. It can also just generate the YAML by passing the `-d / --no-docker-image` flag.